src/app/components/team/Webhooks.js

```diff
@@ -0,0 +1,141 @@
+        <div>
+          <div className={`typography-subtitle2 ${styles['api-keys-heading']}`}>
+            <FormattedMessage
+              defaultMessage="Webhooks [{count}]"
+              description="Title of the webhook management widget"
+              id="webhooks.countHeader"
+              values={{ count: webhooks.edges.length }}
+            />
+            <WebhookEdit />
+          </div>
+          <div className={styles['api-keys']}>
+            {webhooks.edges.length === 0 &&
+              <div className={styles['apikey-entry-root']}>
+                <BlankState noMargin>
+                  <FormattedMessage
+                    defaultMessage="No Webhooks"
+                    description="Label for initial state when there are no API keys created for the current workspace"
+                    id="webhooks.empty"
+                  />
+                </BlankState>
+                <div className={styles['api-keys-empty-blurb']}>
+                  <FormattedMessage
+                    defaultMessage="Add a new integration to be notified of Check events"
+                    description="Subtitle for initial state when there are no webhooks created for the current workspace"
+                    id="webhooks.emptyBlurb"
+                  />
+                </div>
+              </div>
+            }
+            {webhooks.edges.map(w => (<WebhookEntry key={w.node.dbid} webhook={w.node} />))}
+          </div>
+          <div className={`typography-caption ${styles['api-keys-footer']}`}>
+            <HelpIcon />
+            <div>
+              <FormattedMessage
+                defaultMessage="Check out our guide to using the {checkApi} or to troubleshoot common issues."
+                description="Footer text in API key creation widget"
+                id="webhooks.footer"
+                values={{
+                  checkApi: <ExternalLink className="webhooks__api-doc-link" url="https://help.checkmedia.org/en/articles/8773856-check-api-introduction">Check API</ExternalLink>,
+                }}
+              />
+              {' '}
+              { /* The following <a /> element requires Intercom to be configured. Make sure `intercomAppId` is set correctly in config.js  */}
+              { /* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+              <a className="webhooks__intercom-link" href="#" onClick={(e) => { if (Intercom) { Intercom('showNewMessage'); e.preventDefault(); } }}>
+                <FormattedMessage
+                  defaultMessage="Connect with us directly for support."
+                  description="Footer text in API key creation widget"
+                  id="webhooks.footerSupport"
+                />
+              </a>
+            </div>
+          </div>
+        </div>
+      </Collapse>
+    </div>
+  );
+};
+
+Webhooks.propTypes = {
+  team: PropTypes.shape({
+    webhooks: PropTypes.shape({
+      edges: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
```

Comment:
Actually here could be something like:

```javascript
edges: PropTypes.arrayOf(
  PropTypes.exact({
    node: PropTypes.shape({
     dbid: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
).isRequired,
```

Right?

Reply:
`PropTypes.exact` doesn't seem to work quite well with graphql fragment props, as they might contain additional metadata keys which break the exact match:

![image](https://github.com/user-attachments/assets/fbf6d6fe-3956-4f94-982e-57b9aceb7d50)

I'm keeping this as is, as I don't think we need to be too specific here. `PropTypes.object` should be enough when the actual shape is defined in the data fragment.

Reply:
Good point on the meta-fields! Agreed.